File: src/TodoApp.Api/Features/Auth/AuthConstants.cs
namespace TodoApp.Api.Features.Auth;

public static class AuthConstants
{
    // Rate limiting
    public const int RateLimitMaxRequests = 3;
    public const int RateLimitWindowHours = 1;
    
    // Token settings
    public const int TokenExpirationMinutes = 15;
    public const int VerificationCodeLength = 6;
    public const int VerificationCodeMin = 100000;
    public const int VerificationCodeMax = 999999;
    
    // Error titles
    public const string InvalidEmailTitle = "Invalid email";
    public const string TooManyRequestsTitle = "Too many requests";
    public const string InvalidCodeTitle = "Invalid code";
    public const string NoTeamFoundTitle = "No team found";
    public const string UserExistsTitle = "User exists";
    
    // Error messages
    public const string InvalidEmailMessage = "Please provide a valid email address";
    public const string TooManyRequestsMessage = "You've requested too many codes. Please try again later.";
    public const string InvalidOrExpiredCodeMessage = "The verification code is invalid or has expired";
    public const string NoTeamMembershipMessage = "User has no team membership";
    public const string UserAlreadyExistsMessage = "An account with this email already exists";
    
    // Success messages
    public const string VerificationCodeSentMessage = "Verification code sent to your email";
    
    // Email settings
    public const string VerificationEmailSubject = "Your TodoApp verification code";
    public const string WelcomeEmailSubject = "Welcome to TodoApp!";
    
    // Default role
    public const string AdminRole = "Admin";
    
    // Validation messages
    public const string DisplayNameRequiredMessage = "Display name is required";
    public const string DisplayNameMinLengthMessage = "Display name must be at least 2 characters";
    public const string DisplayNameMaxLengthMessage = "Display name must not exceed 100 characters";
    public const string TeamNameRequiredMessage = "Team name is required";
    public const string TeamNameMinLengthMessage = "Team name must be at least 2 characters";
    public const string TeamNameMaxLengthMessage = "Team name must not exceed 100 characters";
    
    // Validation limits
    public const int NameMinLength = 2;
    public const int NameMaxLength = 100;
}

File: src/TodoApp.Web/src/constants/auth.constants.ts
// Auth constants matching backend where applicable

export const AUTH_CONSTANTS = {
  // API endpoints
  API_BASE: '/auth',
  REQUEST_CODE_ENDPOINT: '/auth/request-code',
  VERIFY_CODE_ENDPOINT: '/auth/verify-code',
  COMPLETE_REGISTRATION_ENDPOINT: '/auth/complete-registration',
  
  // Validation
  VERIFICATION_CODE_LENGTH: 6,
  VERIFICATION_CODE_PATTERN: /^\d{6}$/,
  EMAIL_PATTERN: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
  NAME_MIN_LENGTH: 2,
  NAME_MAX_LENGTH: 100,
  
  // Timing
  CODE_EXPIRATION_MINUTES: 15,
  RESEND_COOLDOWN_SECONDS: 60,
  
  // Rate limiting
  RATE_LIMIT_MAX_REQUESTS: 3,
  RATE_LIMIT_WINDOW_HOURS: 1,
  
  // Messages
  VERIFICATION_CODE_SENT: 'Verification code sent to your email',
  INVALID_EMAIL: 'Please provide a valid email address',
  INVALID_CODE: 'The verification code is invalid or has expired',
  TOO_MANY_REQUESTS: "You've requested too many codes. Please try again later.",
  ACCOUNT_EXISTS: 'An account with this email already exists',
  
  // Storage keys
  AUTH_TOKEN_KEY: 'auth_token',
  PENDING_EMAIL_KEY: 'pending_email',
  
  // Roles
  ADMIN_ROLE: 'Admin',
  MEMBER_ROLE: 'Member',
} as const;